packages/edit-site/src/plugins/site-export.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import downloadjs from 'downloadjs';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { MenuItem } from '@wordpress/components';
+import apiFetch from '@wordpress/api-fetch';
+import { download } from '@wordpress/icons';
+import { useDispatch } from '@wordpress/data';
+import { store as noticesStore } from '@wordpress/notices';
+
+export default function SiteExport() {
+	const { createErrorNotice } = useDispatch( noticesStore );
+
+	async function handleExport() {
+		try {
+			const response = await apiFetch( {
+				path: '/wp-block-editor/v1/export',
+				parse: false,
+			} );
+			const blob = await response.blob();
+
+			downloadjs( blob, 'edit-site-export.zip', 'application/zip' );
+		} catch ( error ) {
+			const errorMessage =
+				error.message && error.code !== 'unknown_error'
```

Comment:
There are mutiple error code that could be returned here. 

- rest_cannot_export_templates - permission error. 
- missing_zip_package - Unable to generate ZIP because of missing package. 
- unable_to_create_zip - Unable to generate ZIP for unknown reasons. 

Should there be different error messages here. 

See https://github.com/WordPress/wordpress-develop/pull/2008

Reply:
@spacedmonkey, this is for default error message if the code is unknown. Otherwise, it will use messages returned from the server.